homeassistant/components/http/ban.py

```diff
@@ -249,8 +282,28 @@ def _add_ban(self, ip_ban: IpBan) -> None:
             # Write in a single write call to avoid interleaved writes
             out.write("\n" + yaml.dump(ip_))
 
+    def _write_bans(self, ip_bans: dict[IPv4Address | IPv6Address, IpBan]) -> None:
+        """Write IP bans to config file."""
+        to_write = {
+            str(ip_ban.ip_address): {ATTR_BANNED_AT: ip_ban.banned_at.isoformat()}
+            for ip_ban in ip_bans.values()
+        }
+
+        with open(self.path, "r+", encoding="utf8") as out:
+            out.seek(0)
+            out.truncate()
+            out.write("\n" + yaml.dump(to_write))
```

Comment:
`dict`s are unordered, so this is going to lose ordering of all bans in a file.

I realize you're using a `dict` here because the original structure is a `dict` and then using `yaml.dump`, however order could be preserved by instead sorting and generating each line as it's own dict, similar to how add works.

```suggestion
        to_write = (
            {str(ip_ban.ip_address): {ATTR_BANNED_AT: ip_ban.banned_at.isoformat()}
            for ip_ban in sorted(ip_bans.values(), key=lambda b: b.banned_at)
        )

        with open(self.path, "r+", encoding="utf8") as out:
            out.seek(0)
            out.truncate()
            for ban in to_write:
                out.write("\n" + yaml.dump(ban))
```

Reply:
What do you think about using `OrderedDict` instead?

Reply:
@ViViDboarder Python dicts are not unordered, are you referring to yaml dicts?

Reply:
Oh, interesting. Looks like as of Python 3.7, dictionaries preserve insertion order like `OrderedDict`. I wasn't aware!

Does `yaml.load` preserve the order from the YAML file? If so, this should not be a concern.